plugins/modules/net_tools/nmcli.py

```diff
@@ -1076,7 +1081,7 @@ def connection_update(self, nmcli_command):
                     cmd += ['802-11-wireless.%s' % name, value]
             if self.wifi_sec:
                 for name, value in self.wifi_sec.items():
-                    cmd += ['wifi-sec.%s' % name, value]
+                    cmd += ['802-11-wireless-security.%s' % name, value]
```

Comment:
As per your [suggestion](https://github.com/ansible-collections/community.general/issues/3145#issuecomment-894003609) it probably makes sense to remove the argument duplication in this PR. Better to restrict the bugfix to just passing the password via `stdin`